pyccel/codegen/printing/cucode.py

```diff
@@ -130,4 +147,33 @@ def _print_ModuleHeader(self, expr):
                           global_variables,
                           function_declaration,
                           "#endif // {name.upper()}_H\n"))
+    def _print_Allocate(self, expr):
+        variable = expr.variable
+        shape = ", ".join(self._print(i) for i in expr.shape)
+        if isinstance(variable.class_type, CudaArrayType):
+            dtype = self.find_in_ndarray_type_registry(variable.dtype)
+        elif isinstance(variable.class_type, HomogeneousContainerType):
+            dtype = self.find_in_ndarray_type_registry(numpy_precision_map[(variable.dtype.primitive_type, variable.dtype.precision)])
+        else:
+            raise NotImplementedError(f"Don't know how to index {variable.class_type} type")
+        shape_dtype = self.get_c_type(NumpyInt64Type())
+        shape_Assign = "int64_t shape_Assign [] = {" + shape + "};\n"
+        is_view = 'false' if variable.on_heap else 'true'
+        memory_location = expr.variable.memory_location
+        if memory_location in ('device', 'host'):
+            memory_location = 'allocateMemoryOn' + str(memory_location).capitalize()
+        else:
+            memory_location = 'managedMemory'
+        self.add_import(c_imports['cuda_ndarrays'])
+        self.add_import(c_imports['ndarrays'])
+        alloc_code = f"{self._print(expr.variable)} = cuda_array_create({variable.rank},  shape_Assign, {dtype}, {is_view},{memory_location});\n"
+        return f'{shape_Assign} {alloc_code}'
```

Comment:
I am surprised that the allocation of NDArrays and Cuda arrays is so similar that the allocation command is the same and both imports are needed even when cuda arrays are not used. Probably this is incorrect. I imagine you need something like:
```python
def _print_Allocate(self, expr):
     variable = expr.variable
    if not isinstance(variable.class_type, CudaArrayType):
        return super()._print_Allocate(expr)

    ....
```

Reply:
https://github.com/pyccel/pyccel-cuda/pull/64/commits/96c3f292f8532f0f396bf8016af9f0f9cc6e8ea3